src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -206,7 +209,41 @@ class AppsController(
       throw ex
   }
 
-  private def putSingle(appId: PathId)(implicit identity: Identity): Route = ???
+  private def putSingle(appId: PathId)(implicit identity: Identity): Route =
+    parameter('partialUpdate.as[Boolean].?(true)) { partialUpdate =>
+      update(appId, partialUpdate = partialUpdate, allowCreation = true)
+    }
+
+  /**
+    * Internal representation of `replace or update` logic.
+    *
+    * @param appId appId
+    * @param partialUpdate partial update?
+    * @param allowCreation is creation allowed?
+    * @param identity implicit identity
+    * @return http servlet response
+    */
+  private[this] def update(appId: PathId, partialUpdate: Boolean, allowCreation: Boolean)(implicit identity: Identity): Route = {
+    val version = clock.now()
+
+    (forceParameter &
+      extractClientIP &
+      extractUri &
+      entity(as(appUpdateUnmarshaller(appId, partialUpdate)))) { (force, remoteAddr, requestUri, appUpdate) =>
+        //         Note - this function throws exceptions and handles authorization synchronously. We need to catch and map these
+        //         exceptions to the appropriate rejections
+        val fn = updateOrCreate(
+          appId, _: Option[AppDefinition], appUpdate, partialUpdate, allowCreation, clock.now(), marathonSchedulerService)
```

Comment:
Hm, this is an interesting pattern. Why not create a proper method with `def`? What is the motivation for using the underscore?